### PR TITLE
cloudcheck: add duration param to cloud connection tester

### PR DESCRIPTION
**Pull request by dt**
Release note: none.
Epic: none.

### CODE REVIEW

File: pkg/cloud/cloudcheck/cloudcheck_processor.go [hunk 1]
```diff
@@ -52,24 +65,49 @@ func checkURI(
...
+			return total
+		}
+		defer store.Close()
+
+		res, err := checkStorage(ctx, store, transferSize)
+		if err != nil {
+			res.error = err.Error()
+		}
+
+		if total.ok {
```
Comment (benbardin) [inline pkg/cloud/cloudcheck/cloudcheck_processor.go:99]: If it's not OK, shouldn't we return early? Or at least store the error message? Overwriting seems not quite right

File: docs/generated/sql/bnf/stmt_block.bnf [hunk 2]
```diff
@@ -2696,6 +2699,10 @@ extra_var_value ::=
 show_backup_options_list ::=
 	( show_backup_options ) ( ( ',' show_backup_options ) )*
 
+show_backup_connection_options ::=
```
[the remark this answers is not in the document]
Comment (dt) [reply docs/generated/sql/bnf/stmt_block.bnf:2702]: backup options are formal syntax too these days.

I stuck to all already defined keywords so I don't feel too bad about using syntax (using real syntax also makes it tab complete and self-documenting which seems nice)

File: pkg/sql/execinfrapb/processors_bulk_io.proto [hunk 3]
```diff
@@ -422,8 +422,16 @@ message GenerativeSplitAndScatterSpec {
...
 
+
+
 message CloudStorageTestSpec {
   optional string location = 1 [(gogoproto.nullable) = false];
-  optional int64 transfer_size = 2 [(gogoproto.nullable) = false];
-  // NEXT ID: 2;
+  message Params {
+    // TransferSize determines the size of the file that is written/read per test.
+    optional int64 transfer_size = 1 [(gogoproto.nullable) = false];
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/sql/execinfrapb/processors_bulk_io.proto:432]: Added a comment

File: pkg/cloud/cloudcheck/cloudcheck_stmt.go [hunk 4]
```diff
@@ -31,15 +31,18 @@ var Header = colinfo.ResultColumns{
 	{Name: "locality", Typ: types.String},
 	{Name: "ok", Typ: types.Bool},
 	{Name: "error", Typ: types.String},
+	{Name: "transferred", Typ: types.String},
 	{Name: "read_speed", Typ: types.String},
 	{Name: "write_speed", Typ: types.String},
 	{Name: "can_delete", Typ: types.Bool},
 }
 
+type TestParams = execinfrapb.CloudStorageTestSpec_Params
```
[the remark this answers is not in the document]
Comment (dt) [reply pkg/cloud/cloudcheck/cloudcheck_stmt.go]: Renamed to `Params` (yay for small packages)

File: pkg/cloud/cloudcheck/cloudcheck_processor.go (hunk 1 above)
Comment (dt) [reply pkg/cloud/cloudcheck/cloudcheck_processor.go]: we do return early on the next line when ok is false after we set it here, so we're never overwriting anything other than a zero-value here. restructured it to be more explicit.

File: docs/generated/sql/bnf/stmt_block.bnf (hunk 2 above)
[the remark this answers is not in the document]
Comment (benbardin) [reply docs/generated/sql/bnf/stmt_block.bnf:2702]: ok!

**Review by benbardin — APPROVED** [review]
thanks!

**Comment by dt** [comment]
TFTR!

bors r+

**Pull request merged**